### PR TITLE
marc21: addition of marc21 rule for conference URL

### PR DESCRIPTION
**Pull request by krzysztof**
Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

**Comment by krzysztof** [comment]
@lnielsen Sorry, I thought this was merged already and pushed the files CLI to my master (which is this PR)

File: zenodo/modules/utils/cli.py [hunk 1]
```diff
@@ -26,28 +26,181 @@
...
+    """Add a new file to a publishd record."""
+    pid, record = record_resolver.resolve(recid)
+    bucket = record.files.bucket
+    key = os.path.basename(fp.name)
+
+    obj = ObjectVersion.get(bucket, key)
+    if obj is not None:
+        click.echo('File "{key}" alreay exists in the record {recid}'.format(
+            key=key, recid=recid))
+        return
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: The primary use case is to update an existing file. So perhaps add cli option ``-f/--force`` which will overwrite an existing file.

File: zenodo/modules/utils/cli.py [hunk 2]
```diff
@@ -26,28 +26,181 @@
...
+
+    fp.seek(SEEK_SET, SEEK_END)
+    size = fp.tell()
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: I think ``fp.size`` should work so there's not need to seek in the file.

File: zenodo/modules/utils/cli.py [hunk 3]
```diff
@@ -26,28 +26,181 @@
...
+    click.echo('Will add the following file:\n')
+    click.echo(click.style(
+        '  key: "{key}"\n'
+        '  bucket: {bucket}\n'
+        '  size: {size}\n'
+        ''.format(
+            key=key,
+            bucket=bucket.id,
+            size=size),
+        fg='red'))
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: ```red`` I would only use for actual errors. Here it's information.

File: zenodo/modules/utils/cli.py [hunk 4]
```diff
@@ -26,28 +26,181 @@
...
+        click.echo('File "{key}" not found in the record {recid}'.format(
+            key=key, recid=recid))
+        return
+
+    if click.confirm('Will rename "{key}" to "{new_key}" on bucket {bucket}.'
+                     ' Continue?'.format(
+                        key=obj.key, new_key=new_key, bucket=bucket.id)):
+        record.files.bucket.locked = False
+
+        file_ = obj.file
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: ``file_id = obj.file_id``

File: zenodo/modules/utils/cli.py [hunk 5]
```diff
@@ -26,28 +26,181 @@
...
+        ObjectVersion.delete(bucket, obj.key)
+        ObjectVersion.create(bucket, new_key, _file_id=file_.id)
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: ``..., _file_id=file_id``

File: zenodo/modules/utils/cli.py (hunk 2 above)
Comment (krzysztof) [inline zenodo/modules/utils/cli.py]: No, as it an instance of io.BufferedReader

File: zenodo/modules/utils/cli.py (hunk 3 above)
Comment (krzysztof) [inline zenodo/modules/utils/cli.py]: Changed to green

**Pull request merged**